### PR TITLE
Upgrade MPT to 4.0

### LINKED ISSUE #156: Update Monorepo and MPT Dependencies
Complementary to #149 

Now that the `beta.1` releases on the monorepo are out along https://github.com/ethereumjs/ethereumjs-vm/pull/922 the respective dependencies can be updated here on the client side (see [package.json](https://github.com/ethereumjs/ethereumjs-client/blob/master/package.json) for an overview). This comes along with the patricia tree (MPT) library to also to be updated.

Following updates necessary:

- [x] Replace [ethereumjs-account](https://github.com/ethereumjs/ethereumjs-vm/pull/922) dependency with the `Util.Account` class from [ethereumjs-util](https://github.com/ethereumjs/ethereumjs-util)
- [x] Update [ethereumjs-blockchain](https://github.com/ethereumjs/ethereumjs-blockchain) `v4.0.4` to `@ethereumjs/blockchain` [v5.0.0-beta.1](https://github.com/ethereumjs/ethereumjs-vm/releases/tag/%40ethereumjs%2Fblockchain%405.0.0-beta.1) or subsequent
- [x] Update [ethereumjs-block](https://github.com/ethereumjs/ethereumjs-block) `v2.0.2` to `@ethereumjs/block` [v3.0.0-beta.1](https://github.com/ethereumjs/ethereumjs-vm/releases/tag/%40ethereumjs%2Fblock%403.0.0-beta.1) or subsequent
- [x] Update [ethereumjs-common](https://github.com/ethereumjs/ethereumjs-common)`v1.5.2` to `@ethereumjs/common` [v2.0.0](https://github.com/ethereumjs/ethereumjs-vm/releases/tag/%40ethereumjs%2Fcommon%402.0.0-beta.1) or subsequent
- [ ] update [merkle-patricia-tree](https://github.com/ethereumjs/merkle-patricia-tree) `v3.0.0` to `v4.0.0`

### PR DESCRIPTION
**Pull request by ryanio**
This PR upgrades the `merkle-patricia-tree` dependency to `4.0.0`, simplifies some code in `parse.ts`, and renames `test/util/parse.ts` to `test/util/parse.spec.ts`.

Closes #156 along with #158.

### CODE REVIEW

File: lib/util/parse.ts [hunk 1]
```diff
@@ -44,87 +40,83 @@ export function parseTransports(transports: any[]) {
...
+    await trie.put(address, account.serialize())
   }
-  await Promise.all(promises)
   return trie
 }
 
 async function parseGethHeader(json: any) {
-  return BlockHeader.fromHeaderData(
-    {
-      gasLimit: new util.BN(util.stripHexPrefix(json.gasLimit), 16),
```
Comment (ryanio) [inline lib/util/parse.ts:97]: note: `fromHeaderData` accepts `BufferLike` and `BNLike` instances and converts them, so no need to do it here.

Comment (cgewecke) [reply lib/util/parse.ts:97]: Oh this is a million times nicer! Thought everything needed to be pre-formatted.

Comment (ryanio) [reply lib/util/parse.ts:97]: great! yeah the idea is that the constructor itself (`new BlockHeader()`) is very strict in its types to keep the initialization logic as simple as possible, but the factory helper `fromHeaderData` has flexible input types and does some minimal conversion :)

the same is true with block, tx, account, etc.

**Review by cgewecke — APPROVED** [review]
This all looks really good to me. Super clean. 💯 

(Checked the `.cli` locally and it works)

**Comment by ryanio** [comment]
thanks!

**Pull request merged**